connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -393,13 +394,33 @@ export async function syncOneFile({
     }
   }
 
-  let documentSection: CoreAPIDataSourceDocumentSection | null = null;
+  let documentSection: CoreAPIDataSourceDocumentSection | null;
   if (
     mimeType ===
     "application/vnd.openxmlformats-officedocument.presentationml.presentation"
   ) {
     const data = Buffer.from(downloadRes.data);
-    documentSection = await handlePptxFile(data, file.id, localLogger);
+    documentSection = await handlePptxFile(
+      data,
+      file.id,
+      localLogger,
+      maxDocumentLen
+    );
+  } else if (mimeType === "application/vnd.ms-excel") {
```

Comment:
would it make sense to move that close to the syncSpreadsheet, and have it return early? since they're ~same use case, and both do not require a documentSection?

Reply:
Definitely! I have implemented this logic in a pending GDrive refactoring PR.

Reply:
@philipperolet FYI the GDrive refacto PR is here: https://github.com/dust-tt/dust/pull/6210
If it's okay with you I would wait for this PR to be merged and that we add the missing files support to Microsoft to refacto Microsoft and share resources between GDrive & Microsoft.